core/src/main/java/org/helsinki/paytrail/mapper/PaytrailPaymentCreateResponseMapper.java

```diff
@@ -0,0 +1,32 @@
+package org.helsinki.paytrail.mapper;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.helsinki.paytrail.response.PaytrailResponse;
+import org.helsinki.paytrail.response.payments.PaytrailPaymentCreateResponse;
+
+
+public class PaytrailPaymentCreateResponseMapper extends AbstractModelMapper<PaytrailResponse, PaytrailPaymentCreateResponse> {
+
+    public PaytrailPaymentCreateResponseMapper(ObjectMapper mapper) {
+        super(mapper, PaytrailResponse::new, PaytrailPaymentCreateResponse::new);
+    }
+
+    @Override
+    public PaytrailPaymentCreateResponse to(PaytrailResponse paytrailResponse) {
+        PaytrailPaymentCreateResponse to = super.to(paytrailResponse);
+        try {
+            to.setPaymentResponse(mapper.readValue(paytrailResponse.getResultJson(), new TypeReference<>() {
+            }));
+        } catch (JsonProcessingException e) {
+            String[] errors = new String[]{
+                    e.getMessage(),
+                    e.getOriginalMessage(),
+                    paytrailResponse.getResultJson()
+            };
+            to.setErrors(errors);
```

Comment:
Should you log this error too?

Reply:
Well the errors are added to the returned response object, but sure I could add some logging also.